### PR TITLE
refactor: updated even further dependencies

### PR DESCRIPTION
**Pull request by mfranzke**
### Summary of changes:

Please find the list of all of the changed dependencies with their related breaking changes for all major release updates listed below them. I've as well compared the handlebars and twig `build` output to ensure that this update is fine.

#### Simply updated

- [ejs](https://github.com/mde/ejs/blob/main/CHANGELOG.md)
  - [x] _Removed require.extensions (@mde) / https://github.com/mde/ejs/commit/7c15b781638f775561134eb608ecb9cfe2f761f2_
  Not used by us
  - [x] _Removed legacy preprocessor include (@mde) / [mde/ejs@c6bca15](https://github.com/mde/ejs/commit/c6bca15b0ac56f7e9f97a15afae4dd9a759dc1e0) & [mde/ejs@e6fd45d](https://github.com/mde/ejs/commit/e6fd45d2c48be1c379a6cdb29c5af3527bf1f245)_
  Not used by us
  - [x] _Removed support for EOL Nodes 4 and 6 (@mde)_
  Not relevant for us, as this is out of (below) our Node version scope.
- [scroll-js](https://github.com/markcellus/scroll-js/releases)
  - [x] No documented breaking changes, only _"ESM and CommonJS updates ([#43](https://github.com/markcellus/scroll-js/pull/43)) (https://github.com/markcellus/scroll-js/commit/fe00cc396c67ff099261b0d6e025600a0a743809)"_
  But we need to fix the version at `3.4.x` as with version `3.5.0` a breaking change of jumping from node version 12 to 16 got introduced.
- [classnames](https://github.com/JedWatson/classnames/blob/master/HISTORY.md)
  - [x] minor version update
- [gulp](https://github.com/gulpjs/gulp/releases)
  - [x] _Replace [Orchestrator](https://www.npmjs.com/package/orchestrator) with [Undertaker](https://www.npmjs.com/package/undertaker) ([9fda7b4](https://github.com/gulpjs/gulp/commit/9fda7b4))_ 
  changes documented within the API with commit https://github.com/gulpjs/gulp/commit/0f3151e; no noticeable changes for our implementation
  - [x] _Replace vinyl-fs watch/gaze with chokidar (https://github.com/gulpjs/gulp/commit/2cd0e1e)_
  Not used by our codebase.
  - [x] _Only support tasks that are functions in gulp.watch (https://github.com/gulpjs/gulp/commit/1d70cfb)_
  Not used by our codebase.
  - [x] _Remove array & string task support from gulp.watch (https://github.com/gulpjs/gulp/commit/b085e95)_
  Not used by our codebase.
- [@auto-it/released](https://github.com/intuit/auto/blob/main/CHANGELOG.md)
  - [x] no breaking changes for this package
- [auto](https://github.com/intuit/auto/blob/main/CHANGELOG.md)
  - [x] no breaking changes for this package
- [fs-extra](https://github.com/jprichardson/node-fs-extra/blob/master/CHANGELOG.md)
  - [X] _dropped Node v0.10 support._
  fine with our node.js support
  - [X] _Renamed `clobber` to `overwrite`. This affects `copy()`, `copySync()`, and `move()`. [#330](https://github.com/jprichardson/node-fs-extra/pull/330), [#333](https://github.com/jprichardson/node-fs-extra/pull/333)_
  we don't use this one
  - [X] _BREAKING: Removed support for Node `v0.12`. The Node foundation stopped officially supporting it on Jan 1st, 2017._
  fine with our node.js support
  - [X] _Remove `walk()` and `walkSync()`. `walkSync()` was only part of `fs-extra` for a little over two months. Use [klaw](https://github.com/jprichardson/node-klaw) instead of `walk()`, in fact, `walk()` was just an alias to klaw. For `walkSync()` use [klaw-sync](https://github.com/mawni/node-klaw-sync). See: [#338](https://github.com/jprichardson/node-fs-extra/issues/338), [#339](https://github.com/jprichardson/node-fs-extra/pull/339)_
  we don't use these ones
  - [x] _Added Promise support. All asynchronous native `fs` methods and `fs-extra` methods now return a promise if the callback is not passed. [#403](https://github.com/jprichardson/node-fs-extra/pull/403)_
  This one was added with `fs-extra` major version 3.0.0 – we're only using the async functions in packages that were on higher versions of that dependency even already, and sync elsewhere.
  - [X] _Removed support for setting the default spaces for `writeJson()`, `writeJsonSync()`, `outputJson()`, & `outputJsonSync()`. This was undocumented. https://github.com/jprichardson/node-fs-extra/pull/402_
  We don't use that undocumented way of defining spaces.
  - [x] _Changed behavior of `throws` option for `readJsonSync()`; now does not throw filesystem errors when `throws` is `false`._
  we don't use this one
  - [x] _`writeJson()`, `writeJsonSync()`, `outputJson()`, & `outputJsonSync()` now output minified JSON by default for consistency with `JSON.stringify()`; set the `spaces` option to `2` to override this new behavior. https://github.com/jprichardson/node-fs-extra/pull/402_
  We don't use `fs.spaces`, that has been removed there.
  - [x] _The promisified versions of `fs.read()` & `fs.write()` now return objects. See [the docs](https://github.com/jprichardson/node-fs-extra/blob/master/docs/fs-read-write.md) for details. [#436](https://github.com/jprichardson/node-fs-extra/issues/436), [#449](https://github.com/jprichardson/node-fs-extra/pull/449)_
  we don't use these ones
  - [x] _Significant refactor of `copy()` & `copySync()`, including breaking changes_
  We don't use these `filter` options
    - [x] _The `filter` option can no longer be a RegExp (must be a function). This was deprecated since fs-extra v1.0.0. [#512](https://github.com/jprichardson/node-fs-extra/pull/512)_
    - [x] _`copy()`'s `filter` option can now be a function that returns a Promise. [#518](https://github.com/jprichardson/node-fs-extra/pull/518)_
  - [X] _Drop support for Node.js versions 4, 5, & 7 ([#564](https://github.com/jprichardson/node-fs-extra/pull/564))_
  fine with our node.js support
  - [x] _Refine `copy*()` handling of symlinks to properly detect symlinks that point to the same file. ([#582](https://github.com/jprichardson/node-fs-extra/pull/582))_
  To my knowledge we don't use symlinks
  - [x] _Requires Node.js version 10 or greater ([#725](https://github.com/jprichardson/node-fs-extra/issues/725), [#751](https://github.com/jprichardson/node-fs-extra/pull/751))_
  fine with our node.js support
  - [x] _Switched `ensureDir*` to use a fork of https://github.com/sindresorhus/make-dir to make use of native recursive `fs.mkdir` where possible ([#619](https://github.com/jprichardson/node-fs-extra/issues/619), [#756](https://github.com/jprichardson/node-fs-extra/pull/756))_
  Shouldn't be a problem for us. In [detail](https://github.com/jprichardson/node-fs-extra/pull/756):
    - [x] _We no longer return a file path on success (to match `fs.mkdir`)_
    - [x] _We do not allow creating root on Windows (matches `fs.mkdir`)_
    - [x] _Error codes are a bit different_
  - [x] _Properly preserve `atime` for `copy*` with `preserveTimestamps` option ([#633](https://github.com/jprichardson/node-fs-extra/pull/633))_
  We don't use this option.
  - [x] _`outputJson` now outputs objects as they were when the function was called, even if they are mutated later ([#702](https://github.com/jprichardson/node-fs-extra/issues/702), [#768](https://github.com/jprichardson/node-fs-extra/pull/768))_
  I can't see how this would affect us negatively.
  - [x] _Cannot pass `null` as an options parameter to `*Json*` methods ([#745](https://github.com/jprichardson/node-fs-extra/issues/745), [#768](https://github.com/jprichardson/node-fs-extra/pull/768))_
  We don't do this.
  - [x] _Require Node.js v12+ ([#886](https://github.com/jprichardson/node-fs-extra/issues/886), [#893](https://github.com/jprichardson/node-fs-extra/pull/893), [#890](https://github.com/jprichardson/node-fs-extra/pull/890), [#894](https://github.com/jprichardson/node-fs-extra/pull/894), [#895](https://github.com/jprichardson/node-fs-extra/pull/895))_
  fine with our node.js support
  - [x] _Allow copying broken symlinks ([#779](https://github.com/jprichardson/node-fs-extra/pull/779), [#765](https://github.com/jprichardson/node-fs-extra/issues/765), [#638](https://github.com/jprichardson/node-fs-extra/issues/638), [#761](https://github.com/jprichardson/node-fs-extra/issues/761))_
  To my knowledge we don't use symlinks
  - [x] _Ensure correct type when destination exists for `ensureLink*()`/`ensureSymlink*()` ([#826](https://github.com/jprichardson/node-fs-extra/pull/826), [#786](https://github.com/jprichardson/node-fs-extra/issues/786), [#870](https://github.com/jprichardson/node-fs-extra/issues/870))_
  We don't use these methods.
  - [x] _Error when attempting to `copy*()` unknown file type ([#880](https://github.com/jprichardson/node-fs-extra/pull/880))_
  I can't see how this would affect us.
  - [x] _Remove undocumented options for `remove*()` ([#882](https://github.com/jprichardson/node-fs-extra/pull/882))_
  We don't use any `remove*()` function
- [better-opn](https://github.com/ExiaSR/better-opn/releases)
  - [x] _Drop support for Node.js version lower than 12_
  fine with our node.js support
  - [x] _The `app` option now accepts an object instead of a string/array:_
  We don't use this option.
- [deepmerge](https://github.com/TehShrike/deepmerge/blob/master/changelog.md)
  - [x] _drop ES module build [#123](https://github.com/TehShrike/deepmerge/issues/123)_
  Fine for us as we're using the Common JS version.
  - [x] _The `main` entry point in `package.json` is now a CommonJS module instead of a UMD module [#155](https://github.com/TehShrike/deepmerge/pull/155)_
  Fine for us as we're using the Common JS version.
- [execa](https://github.com/sindresorhus/execa/releases)
  - [x] _Remove faulty emulated ENOENT error on Windows ([#447](https://github.com/sindresorhus/execa/pull/447)) https://github.com/sindresorhus/execa/commit/bdbd97517122ce7de8a36ca5ce827f86003febf0 This is only a breaking change if you depend on the exact error message._
  We're aren't depending on the exact error message.
- [jsdoc-to-markdown](https://github.com/jsdoc2md/jsdoc-to-markdown/releases)
  - [x] _Node versions less than v10 are no longer supported_
  fine with our node.js support
- [archiver](https://github.com/archiverjs/node-archiver/blob/master/CHANGELOG.md)
  - [x] _follow node LTS, remove support for versions under 6. ([#339](https://github.com/archiverjs/node-archiver/pull/339))_
  fine with our node.js support
  - [x] _slowly catch up with node LTS, remove support for versions under 8._
  fine with our node.js support
  - [x] _absolute path glob patterns are no longer supported: use cwd option instead._
  we do.
- [has-yarn](https://github.com/sindresorhus/has-yarn/releases)
  - [x] Require Node.js 8 https://github.com/sindresorhus/has-yarn/commit/8f6bc3683d582fdb71b668e3b4b38099920d5301
  fine with our node.js support
- [inquirer](https://github.com/SBoudrias/Inquirer.js/releases)
  - [x] _Update to Rx.js v6_
  I can't see how this would affect us.
  - [x] _Drop support for Node 6._
  fine with our node.js support
  - [x] _Drop support for Node 10 (through dependencies)_
  fine with our node.js support
- [ora](https://github.com/sindresorhus/ora/releases)
  - [x] _Require Node.js 6 https://github.com/sindresorhus/ora/commit/17c1a29ef7257d4f4f54029e491cb496c2f83f7e_
  fine with our node.js support
  - [x] _Rename the `enabled` option to `isEnabled`_ https://github.com/sindresorhus/ora/commit/fc5c6607565df17c822d694442389fbc6c4864fa
  We don't use this option.
  - [x] _Ora should still output text when not `iEnabled`_ https://github.com/sindresorhus/ora/commit/d1f8d8529b3f95a28a5b561d29870a2e3250b1c5
  I can't see how this would affect us.
- [yargs](https://github.com/yargs/yargs/blob/main/CHANGELOG.md)
  - [x] _node: drop Node 10 ([#1919](https://github.com/yargs/yargs/pull/1919))_
  fine with our node.js support
  - [x] _implicitly private methods are now actually private_
  I can't see how this would affect us.
  - [x] _deprecated reset() method is now private (call yargs() instead)._
  We don't use this method.
  - [x] _yargs-factory: refactor yargs-factory to use class ([#1895](https://github.com/yargs/yargs/pull/1895))_
  We don't use `yargs-factory`.
  - [x] _.positional() now allowed at root level of yargs._
  We don't use that method.
  - [x] _coerce: coerce is now applied before validation._
  I can't see how this would affect us.
  - [x] _async: yargs now returns a promise if async or check are asynchronous._
  I can't see how this would affect us.
  - [x] _middleware: global middleware now applied when no command is configured._
  I can't see how this would affect us.
  - [x] _[#1823](https://github.com/yargs/yargs/pull/1823) contains the following breaking API changes:_
  I can't see how these would affect us.
    - [x] _now returns a promise if handler is async._
    I can't see how this would affect us.
    - [x] _onFinishCommand removed, in favor of being able to await promise._
    We don't use this one.
    - [x] _getCompletion now invokes callback with err and `completions, returns promise of completions._
    We don't use this one.
- [markdown-it](https://github.com/markdown-it/markdown-it/blob/master/CHANGELOG.md)
  - [x] _Major version bumped, because internals or `linkify-it` was changed. You will not be affected anyhow, if not used direct access to `require('linkify-it/re')` for customizations._
  We don't access `require('linkify-it/re')` directly.
  - [x] _Minor internal structs change, to make pairs parse more effective (cost is linear now). If you use external "pairs" extensions, you need sync those with "official ones". Without update, old code will work, but can cause invalid result in rare case. This is the only reason of major version bump. With high probability you don't need to change your code, only update version dependency._
  I can't see how this would affect us.
- [`send`](https://github.com/pillarjs/send/blob/master/HISTORY.md)
  - [x] no  breaking changes
- [`commander`](https://www.npmjs.com/package/commander)
  - [6.0.0](https://github.com/tj/commander.js/compare/v5.1.0..v6.0.0) (2020-07-21), Added
    - allow options to be added with just a short flag ([#1256](https://github.com/tj/commander.js/pull/1256))
      - [x] _Breaking: the option property has same case as flag. e.g. flag `-n` accessed as `opts().n` (previously uppercase)_
      Not in use by us.
    - [x] _Breaking: throw an error if there might be a clash between option name and a Command property, with advice on how to resolve ([#1275](https://github.com/tj/commander.js/pull/1275))_
    We don't get an error like this.
  - [5.0.0](https://github.com/tj/commander.js/compare/v4.1.1..v5.0.0) (2020-03-14)
    - [x] _Breaking: .args contains command arguments with just recognised options removed ([#1032](https://github.com/tj/commander.js/issues/1032) [#1138](https://github.com/tj/commander.js/issues/1138))_
    Not in use by us
    - [x] _Breaking: display error if required argument for command is missing ([#995](https://github.com/tj/commander.js/issues/995) [#1149](https://github.com/tj/commander.js/issues/1149))_
    - [x] _Breaking: `.allowUnknownOption()` ([#802](https://github.com/tj/commander.js/issues/802) [#1138](https://github.com/tj/commander.js/issues/1138))_
    Not in use by us
      - unknown options included in arguments passed to command action handler
      - unknown options included in `.args`
    - [x] _Breaking: `.parseOptions()` ([#1138](https://github.com/tj/commander.js/issues/1138))_
    Not in use by us
      - `args` in returned result renamed `operands` and does not include anything after first unknown option
      - unknown in returned result has arguments after first `unknown` option including operands, not just options and values
    - _Breaking_ `.on('command:*', callback)` and other command events passed (changed) results from `.parseOptions`, i.e. operands and unknown ([#1138](https://github.com/tj/commander.js/issues/1138))
  - [4.0.0](https://github.com/tj/commander.js/compare/v3.0.2..v4.0.0) (2019-11-02)
    - Changed
      - [x] _Breaking: Commander is only officially supported on Node 8 and above, and requires Node 6 ([#1053](https://github.com/tj/commander.js/issues/1053))_
      We're on Node 12
    - Fixed
      - _Breaking_ keep command object out of program.args when action handler called ([#1048](https://github.com/tj/commander.js/issues/1048))
        - also, action handler now passed array of unknown arguments
  - [3.0.0](https://github.com/tj/commander.js/compare/v2.20.1...v3.0.0) / 2019-08-08
    - _Breaking_ Change TypeScript to use overloaded function for `.command`. ([#938](https://github.com/tj/commander.js/issues/938) [#990](https://github.com/tj/commander.js/issues/990))
    - Fix behavior of --no-* options ([#795](https://github.com/tj/commander.js/issues/795))
      - _Breaking_ custom event listeners: `--no-foo` on cli now emits `option:no-foo` (previously `option:foo`)
      - _Breaking_ default value: defining `--no-foo` after defining `--foo` leaves the default value unchanged (previously set it to false)

#### Updated and migrated some code

- [cosmiconfig](https://github.com/davidtheclark/cosmiconfig/blob/main/CHANGELOG.md):
> Breaking change: Separate async and sync APIs, accessible from different named exports. If you used `explorer.searchSync()` or `explorer.loadSync()`, you'll now create a sync explorer with `cosmiconfigSync()`, then use `explorerSync.search()` and `explorerSync.load()`.
- [fuse.js](https://github.com/krisk/Fuse/blob/master/CHANGELOG.md):
> Removed tokenization [Discussion](https://github.com/krisk/Fuse/issues/355)
> Removed maxPatternLength option
- [js-yaml](https://github.com/nodeca/js-yaml/blob/master/CHANGELOG.md): [Migration guide](https://github.com/nodeca/js-yaml/blob/master/migrate_v3_to_v4.md)
`safeLoad()`, `safeLoadAll()`, `safeDump()` => `load()`, `loadAll()`, `dump()`
- [husky](https://github.com/typicode/husky/releases/tag/v7.0.0)
  - [x] _Improve `.husky/` directory structure. `.husky/.gitignore` is now unnecessary and can be removed._
  Did that.
  - [x] _Drop Node 10 support_
  Not relevant for us, as we're on Node 12

#### Replaced

- [opn](https://www.npmjs.com/package/opn): [`opn`](https://www.npmjs.com/package/opn) got renamed to [`open`](https://www.npmjs.com/package/open) (still maintained by @sindresorhus).

#updatingdependencies[bit](https://github.com/pattern-lab/patternlab-node/pull/1299)by[bit](https://github.com/pattern-lab/patternlab-node/pull/1300)by[bit](https://github.com/pattern-lab/patternlab-node/pull/1302)by[bit](https://github.com/pattern-lab/patternlab-node/pull/1301)by[bit](https://github.com/pattern-lab/patternlab-node/pull/1305)by[bit](https://github.com/pattern-lab/patternlab-node/pull/1306)by[bit](https://github.com/pattern-lab/patternlab-node/pull/1307)by[bit](https://github.com/pattern-lab/patternlab-node/pull/1303)

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -12,17 +12,17 @@
...
+    "execa": "5.0.0",
+    "fs-extra": "10.0.0",
     "glob": "7.1.6",
-    "has-yarn": "1.0.0",
-    "inquirer": "5.1.0",
+    "has-yarn": "2.1.0",
+    "inquirer": "8.0.0",
     "lodash": "4.17.21",
-    "ora": "2.1.0",
+    "ora": "5.4.0",
```
Comment (JosefBredereck) [inline packages/cli/package.json]: This is a huge version skip. Aren't there no side effects or breaking changes in the CLI?
This also applies to inquirer with 3 major version increases.

File: packages/engine-mustache/package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "version": "5.14.3",
   "main": "lib/engine_mustache.js",
   "dependencies": {
-    "fs-extra": "0.30.0",
+    "fs-extra": "10.0.0",
```
Comment (JosefBredereck) [inline packages/engine-mustache/package.json:7]: It was time to update that one. 🚀

**Review by JosefBredereck — COMMENTED** [review]
I rechecked the CLI, seems that there are no problems with our current config

File: packages/cli/package.json (hunk 1 above)
Comment (mfranzke) [reply packages/cli/package.json]: I've checked all of those packages release by release within their respective changelogs. On the things that needed an update, I've listed those within the related "Updated and migrated some code" section of the description.

 I would like to suggest to collect the breaking changes for each of those breaking releases and add them in the above listing as well for a better overview.

**Comment by mfranzke** [comment]
extracted recursive-copy update to https://github.com/pattern-lab/patternlab-node/pull/1352

**Review by JosefBredereck — APPROVED** [review]

**Comment by JosefBredereck** [comment]
PR was released with v5.16.0

**Pull request merged**